test/server/ip-tools.js

```diff
@@ -33,6 +33,27 @@ describe("IP tools", () => {
 		assert.deepEqual(cidrRange, stringRange);
 	});
 
+	it('should not parse invalid ranges', () => {
+		assert.equal(IPTools.isValidRange('42.42.10.0 - 42.42.5.0'), false);
+		assert.equal(IPTools.isValidRange('250.0.0.0 - 260.0.0.0'), false);
+		assert.equal(IPTools.isValidRange('250.0.0.0/43'), false);
+		assert.equal(IPTools.isValidRange('250.0.0.0/16x'), false);
+		assert.equal(IPTools.isValidRange('250.0.0.0.1/24'), false);
+		assert.equal(IPTools.isValidRange('250.0.0.0.1 - 250.0.0.2'), false);
+		assert.equal(IPTools.isValidRange('250.0.0.0.*'), false);
+	});
+
+	it('should reject invalid IPs', () => {
+		assert.equal(IPTools.ipToNumber('256.0.0.0'), null);
+		assert.equal(IPTools.ipToNumber('42.0.0.1111'), null);
+		assert.equal(IPTools.ipToNumber('42.0.hi.0'), null);
+		assert.equal(IPTools.ipToNumber('256.0.0.0.1'), null);
+		assert.equal(IPTools.ipToNumber('256.0.0hi.1'), null);
```

Comment:
```suggestion
		assert.equal(IPTools.ipToNumber('256.0.0hi.1'), null);
		assert.equal(IPTools.ipToNumber('256.0.1'), null);
```

Since we've committed not to take shorthand IPs, let's make sure they're `null`.

Reply:
Fair enough.